### DISCUSSION TITLE
Unable to import txt EBSD file

### DISCUSSION THREAD

**Discussion by trudator** [mtex-toolbox/mtex#2160] [Ask Anything]
Hello,

I was unable able to import the txt file of my EBSD data retrieved from an EDAX system. The dataset is also attached. I appreciate your help. Thank you.

![image](https://github.com/mtex-toolbox/mtex/assets/117386810/b3f88426-83e2-4b4b-b1a9-2410bb02b50b)

[15raw.txt](https://github.com/user-attachments/files/15905057/15raw.txt)

**Comment by kilir** [comment]
Hi,

your file contains all sorts of data about grains, but it is not EBSD data. Is that what you really want to load?

```
# Column 1: Integer identifying grain. Negative value indicates an anti-grain
# Column 2: An integer identifying the phase (-1 for anti-grains)
#           1 -  Titanium (Beta)
#           2 -  Titanium (Alpha)
# Column 3-5: Average orientation (phi1, PHI, phi2) in degrees (360, 360, 360 for anti-grains)
# Column 6-8: Average orientation (phi1, PHI, phi2) in radians (2pi, 2pi, 2pi for anti-grains)
# Column 9-10: Average Position (x, y) in microns
# Column 11: Average Image Quality (IQ)
# Column 12: Average Confidence Index (CI, -1 for anti-grains)
# Column 13: Average Fit (degrees, 180 for anti-grains or invalid grains)
# Column 14: Edge grain (1 = touching edge, otherwise 0)
# Column 15: Number of measurement points in grain
# Column 16: Area of grain in square microns
# Column 17: Diameter of grain in microns
...
```

Cheers,

Rüdiger

**Reply by trudator** [reply]
I think it consists all the necessary data like Euler angles and their corresponding x,y, right? This is the one I exported from EDAX.

**Reply by Tijmenvermeij** [reply]
I tihnk Edax typically exports grains to the .txt file once you do a grain reconstruction... Best to just export .ang or maybe .h5. Or try to export before doing a grain reconstruction.
